src/param.cpp

```diff
@@ -83,9 +83,13 @@ bool operator!=(const param &p0, const param &p1)
     return !(p0 == p1);
 }
 
-expression subs(const param &p, const std::unordered_map<std::string, expression> &)
+expression subs(const param &p, const std::unordered_map<std::string, expression> &smap)
 {
-    return expression{p};
+    if (auto it = smap.find("p"+std::to_string(p.idx())); it == smap.end()) {
```

Comment:
Can you use ``_format`` here as well, instead of ``to_string()``?

Reply:
now outdated, but format would not compile?

Reply:
probably forgot the ``using literal`` thing I guess.